authenticate/handlers.go

```diff
@@ -97,7 +97,13 @@ func (a *Authenticate) SignIn(w http.ResponseWriter, r *http.Request) {
 	// original `state` parameter received from the proxy application.
 	state := r.Form.Get("state")
 	if state == "" {
-		httputil.ErrorResponse(w, r, "no state parameter supplied", http.StatusForbidden)
+		httputil.ErrorResponse(w, r, "no state parameter supplied", http.StatusBadRequest)
+		return
+	}
+
+	redirectURI := r.Form.Get("redirect_uri")
+	if redirectURI == "" {
+		httputil.ErrorResponse(w, r, "no redirect_uri parameter supplied", http.StatusBadRequest)
```

Comment:
Could you add a unit test for this case?

Reply:
@desimone  Oh.., sorry, I think I mis-rebased upstream master...It is correct that this diff is not necessary. 
https://github.com/pomerium/pomerium/commit/66b4c2d3cd368b26fdb1ce905d4935f0cf3e940b#diff-d011e1d3c12517c5bd8581ef91d69fa3L107

Reply:
No worries! Thank you 👍